src/ecmascript_simd.js

```diff
@@ -117,7 +117,14 @@ function simdCheckLaneIndex(index, lanes) {
 var lanes = [];
 
 function simdCreate(type) {
-  return type.fn.apply(type.fn, lanes);
+  // Work around v8 NaN canonicalization issue: if lanes contains floats with non-canonical NaN bit patterns,
+  // type.fn.apply() will canonicalize the NaNs and the bits are lost (most likely as part of float->double expansion).
+  // Directly passing the arguments into the function preserves them.
+  if (type.name == "Float32x4") {
+    return SIMD.Float32x4(lanes[0], lanes[1], lanes[2], lanes[3]);
+  } else {
+    return type.fn.apply(type.fn, lanes);
+  }
```

Comment:
This is the fix which is likely to land soon in the upstream version of this file, right? If so, then this is good.